### ISSUE TITLE
[a11y]: Modal [all variants] - VO does not identify the beginning / ending of the Landmark region

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14637]
### Package

@carbon/react

### Browser

Safari & Iphone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO 

### Description
Unable to dismiss the Modal using the Keyboard and also buttons are not operable

1. None of the buttons are operable with in the Modal using the Keyboard. i.e., Close, Cancel, Add buttons.

2. Unable to dismiss the Modal Dialog by Pressing the Escape Key.

NOTE: Because of these issues we are not able to test the Modal Dialog completely hence once these defects are fixed we will continue the testing.

NOTE: As per the discussion in today's call (6th Sep 2023) we understood that there is no functionality implemented for ADD button hence it is not operable .. However we would like to request to implement a sample functionality i.e., opens in a new window which will help us to verify the keyboard operability of the Add button.

### Recommended Fix
All the buttons with in the Modal should be Keyboard operable.
Along with these user should be able to dismiss the Modal Dialog using the Escape Key.

### WCAG 2.1.1 Keyboard 

https://react.carbondesignsystem.com/?path=/story/components-modal--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems](https://github.com/carbon-design-system/carbon/issues/14255)
